executor/sort.go

```diff
@@ -163,6 +294,58 @@ func (e *SortExec) keyColumnsLess(i, j int) bool {
 	return e.lessRow(rowI, rowJ)
 }
 
+func (e *SortExec) readPartition(disk *chunk.ListInDisk, rowPtrs []chunk.RowPtr) error {
+	e.rowChunks = chunk.NewList(retTypes(e), e.initCap, e.maxChunkSize)
+	for _, rowPtr := range rowPtrs {
+		rowPtr, err := disk.GetRow(rowPtr)
+		if err != nil {
+			return err
+		}
+		e.rowChunks.AppendRow(rowPtr)
+	}
+	return nil
+}
+
+// alreadySpilled indicates that records have spilled out into disk.
+func (e *SortExec) alreadySpilled() bool { return e.rowChunksInDisk != nil }
+
+// alreadySpilledSafe indicates that records have spilled out into disk. It's thread-safe.
+func (e *SortExec) alreadySpilledSafe() bool { return atomic.LoadUint32(&e.spilled) == 1 }
```

Comment:
We only need to keep one of them?

Reply:
The function is used for spillDiskAction trigger, so I copy this code from hash_table spill disk. : )